components/resources/library/src/commonTest/kotlin/org/jetbrains/compose/resources/TestUtils.kt

```diff
@@ -66,4 +74,49 @@ internal fun parsePluralSamples(samples: String): List<Int> {
 internal fun pluralRuleListOf(vararg rules: Pair<PluralCategory, String>): PluralRuleList {
     val pluralRules = rules.map { PluralRule(it.first, it.second) } + PluralRule(PluralCategory.OTHER, "")
     return PluralRuleList(pluralRules.toTypedArray())
-}
\ No newline at end of file
+}
+
+/**
+ * Executes a test block within a Compose UI testing environment while ensuring
+ * that any cached resources are cleared before the test begins.
+ */
+@OptIn(ExperimentalTestApi::class)
+fun runComposeResourceTest(block: ComposeUiTest.() -> Unit) = runComposeUiTest {
+    var areCachesCleared = false
+    setContent {
```

Comment:
is this `setContent { LaunchedEffect() ... }` required?

Is it needed to run the coroutine?
____

Do such tests pass on web? The `waitUntil` is blocking the event loop. I'd expect that on web the tests will just timeout.

Reply:
I see. The mutex is not locked, so it proceeds without suspending there. 

Could you please add a TODO to refactor this code after updating to CMP 1.9.0?
The refactoring would get rid of `setContent { LaunchedEffect() ... }` and the coroutine can be launched right in the test block - it will be suspend in 1.9.0.